### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: Cannot set property '$ts' of undefined

### ISSUE THREAD

**Issue by MarcMcIntosh** [solzimer/session-pouchdb-store#1]
I occasionally see this warning... any idea what's causing it?
running with the path to local directory as the first argument  if that makes any difference.

**Comment by solzimer** [comment]
Hi @MarcMcIntosh ,

Sorry for the delay, I've been very busy lately. Could you post how you initialize the store, so I can test what is happening?

**Comment by MarcMcIntosh** [comment]
hi @solzimer 

using webpack btw... not sure if that makes a difference. seems to work with these settings.

```js
const { resolve } = require('path');
require('events').EventEmitter.defaultMaxListeners = 0;
const session = require('express-session');
const PouchSession = require('session-pouchdb-store');

const DATA_DIR = resolve(__dirname, '..', '..', '..', 'data', 'sessions');
const SECRET_KEY = require('./key');
const MAX_AGE = 1000 * 60 * 60 * 24 * 3; // 3 days,
const storeOptions = { maxIdle: MAX_AGE, scavenge: 5 * 60 * 1000, purge: MAX_AGE };

module.exports = session({
  resave: false,
  saveUninitialized: true,
  rolling: true,
  secret: SECRET_KEY,
  cookie: { maxAge: MAX_AGE },
  store: new PouchSession(DATA_DIR, storeOptions),
});

**Comment by gruberjl** [comment]
I'm running into the same error:

Here's the full error:
```
(node:4524) UnhandledPromiseRejectionWarning: TypeError: Cannot set property '$ts' of undefined
    at SessionStore.touch (C:\Users\JohnGruber\Documents\web-server\node_modules\session-pouchdb-store\lib\store.js:193:26)
    at ServerResponse.end (C:\Users\JohnGruber\Documents\web-server\node_modules\express-session\index.js:342:15)
    at ServerResponse.send (C:\Users\JohnGruber\Documents\web-server\node_modules\express\lib\response.js:221:10)
    at ServerResponse.json (C:\Users\JohnGruber\Documents\web-server\node_modules\express\lib\response.js:267:15)
    at update (C:\Users\JohnGruber\Documents\web-server\server\gitbit\api\update.js:18:21)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
(node:4524) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:4524) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The function that seems to be causing the error is:
```
	touch(sid, session, callback) {
		this.sessions[sid].$ts = Date.now();
		callback();
	}
```

I put a couple of console.logs to see a little bit more information and it looked like this:
```
	touch(sid, session, callback) {
		console.log(this.sessions) // returned {}
		console.log(`SID: ${sid}`) // returned 54IK99fc2eooyZd1b78KoDMoijagyBzL
		this.sessions[sid].$ts = Date.now();
		callback();
	}
```

My session options look like this:
```
{
    secret: process.env.secret,
    resave: false,
    saveUninitialized: true,
    cookie: { secure: false },
    store : new PouchSession(new PouchDB('http://localhost:5984/sessions', {}))
}
```

It's happening in my test environment pretty often when I stop and start my server but don't close my browser so I'm worried it will have an issue in production anytime a session is routed to a different backend server.

I checked CouchDb and found the session so I'm guessing touch is being called before getting the sessions from the database?

**Comment by MarcMcIntosh** [comment]
Does `this.sessions[sid]` exist?

**Comment by solzimer** [comment]
Hi,
Sorry, I've been pretty busy lately. I'll take a look into this issue as soon as I can

**Comment by solzimer** [comment]
Hi again,

You were right. Express session API isn't very clear about the "touch" method, so there are situations where server tries to keep alive a session that haven't been fetched from the store yet.

The issue have been resolved in version 0.2.2

Thanks for your feedback, and, again, sorry for the delay.

**Comment by MarcMcIntosh** [comment]
@solzimer 
What was the fix :)
